### PR TITLE
Fix/4876 missing token detection event properties

### PR DESCRIPTION
**Pull request by Cal-L**
**Description**

#4876

Filtered events in MixPanel shows `Tokens Hidden` where the `tokens` field is populated [here](https://mixpanel.com/project/1979229/view/132079/app/events#~(dateRange~(from~'!start_of_current_day~to~'!end_of_current_day~type~'between)~entries~(~(aggregationOperator~'total~aggregationOperatorPerUser~null~dataGroupId~null~event~(value~'!all_events~label~'All*20Events~isRecentlyUsed~false)~filters~(~(resourceType~'event~propertyName~'!app_version_string~propertyObjectKey~null~propertyDefaultType~'string~propertyType~'string~filterOperator~'equals~filterValue~(~'5.5.2))~(resourceType~'event~propertyName~'!app_build_number~propertyObjectKey~null~propertyDefaultType~'string~propertyType~'string~filterOperator~'equals~filterValue~(~'949)))~filtersOperator~'and~property~null~type~'event))~filters~(~)~search~'~colWidths~(*2524event_name~233~*2524time~216~*2524distinct_id~205~*2524city~204~mp_country_code~195~*2524os~179)~selectedProperties~(~(value~'!event_name~resourceType~'event~label~'Event*20Name~type~'string~propertyDefaultType~'string)~(value~'!time~resourceType~'event~label~'!time~type~'string~propertyDefaultType~'string)~(value~'!distinct_id~resourceType~'event~label~'Distinct*20ID~type~'string~propertyDefaultType~'string)~(value~'!city~resourceType~'event~label~'City~type~'string~propertyDefaultType~'string)~(value~'mp_country_code~resourceType~'event~label~'Country~type~'string~propertyDefaultType~'string)~(value~'!os~resourceType~'event~label~'Operating*20System~type~'string~propertyDefaultType~'string))~showQueryBuilder~true))

**Screenshots/Recordings**

_If applicable, add screenshots and/or recordings to visualize the before and after of your change_

**Issue**

Progresses #???

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

File: bitrise.yml [hunk 1]
```diff
@@ -323,10 +323,10 @@ app:
           PROJECT_LOCATION_IOS: ios
         - opts:
               is_expand: false
-          VERSION_NAME: 5.5.0
+          VERSION_NAME: 5.5.2
```
Comment (Cal-L) [inline bitrise.yml:326]: Revert version number before merging

File: bitrise.yml [hunk 2]
```diff
@@ -323,10 +323,10 @@ app:
...
         - opts:
               is_expand: false
-          VERSION_NUMBER: 939
+          VERSION_NUMBER: 949
```
Comment (Cal-L) [inline bitrise.yml:329]: Revert build number before merging

File: android/app/build.gradle [hunk 3]
```diff
@@ -160,8 +160,8 @@ android {
 		applicationId "io.metamask"
 		minSdkVersion rootProject.ext.minSdkVersion
 		targetSdkVersion rootProject.ext.targetSdkVersion
-		versionCode 939
-		versionName "5.5.0"
+		versionCode 949
+		versionName "5.5.2"
```
Comment (gantunesr) [inline android/app/build.gradle:164]: Revert before merge

File: app/util/analyticsV2.js [hunk 4]
```diff
@@ -200,7 +200,11 @@ export const trackEventV2 = (eventName, params) => {
       for (const key in params) {
         const property = params[key];
 
-        if (property && typeof property === 'object') {
+        if (
+          property &&
+          typeof property === 'object' &&
+          !Array.isArray(property)
+        ) {
```
Comment (gantunesr) [inline app/util/analyticsV2.js:207]: Are you sure this is not breaking any other event?

**Review by gantunesr — CHANGES_REQUESTED** [review]
LGTM. But it's required to revert the version and build number before merge

File: app/util/analyticsV2.js (hunk 4 above)
Comment (Cal-L) [reply app/util/analyticsV2.js:207]: Yep. We don't have any events where we're sending a list format, so this logic wouldn't apply to any post change analytics events. However, this was added to support list format for mix panel.

**Review by gantunesr — APPROVED** [review]

**Pull request merged**